File: Java/src/main/java/uk/org/winder/pi_quadrature/Output.java
/*
 *  Output functions for π using quadrature codes.
 *
 *  Copyright © 2009–2013  Russel Winder
 */

package uk.org.winder.pi_quadrature;

public class Output {

  public static void out(final String prefix, final Double pi, final Integer n, final Double elapseTime) {
    System.out.println("====================== " + prefix);
    System.out.println("\tπ = " + pi);
    System.out.println("\titeration count = " + n);
    System.out.println("\telapse time = " + elapseTime);
  }

  public static void out(final String prefix, final Double pi, final Integer n, final Double elapseTime, final Integer numberOfTasks) {
    out(prefix + ": task count = " + numberOfTasks.toString(), pi, n, elapseTime);
    System.out.println("\tprocessor count = " + Runtime.getRuntime().availableProcessors());
  }

}

File: Kotlin/src/main/kotlin/uk/org/winder/pi_quadrature/output.kt
/*
 *  Output functions for calculations of π using quadrature.
 *
 *  Copyright © 2013–2015  Russel Winder
 */

package uk.org.winder.pi_quadrature

fun out(banner:String, pi:Double, n:Int, elapseTime:Double) {
	println("============================ " + banner)
	println("\tπ = " + pi)
	println("\titeration count = " + n)
	println("\telapse time = " + elapseTime)
}

fun out(banner:String, pi:Double, n:Int, elapseTime:Double, numberOfTasks:Int) {
	out(banner + ": task count" + numberOfTasks, pi, n, elapseTime)
	println("\tprocessor count = " + Runtime.getRuntime().availableProcessors())
}
